### ISSUE TITLE
Unable to open 'file.ext': File not found (file:///file.ext).

### ISSUE THREAD

**Issue by sebasijan** [dotnet/vscode-csharp#2374]
## Environment data
`dotnet --info` output:

	.NET Command Line Tools (2.1.300-preview1-008174)

	Product Information:
	 Version:            2.1.300-preview1-008174
	 Commit SHA-1 hash:  b8df89a54f

	Runtime Environment:
	 OS Name:     Windows
	 OS Version:  10.0.17134
	 OS Platform: Windows
	 RID:         win10-x64
	 Base Path:   C:\Program Files\dotnet\sdk\2.1.300-preview1-008174\

	Microsoft .NET Core Shared Framework Host

	  Version  : 2.1.0-preview1-26216-03
	  Build    : f2c3216183d20416568a4bbf5bb7d153e826f153

VS Code version:

	Visual Studio Code
	Version 1.23.1
	Commit d0182c3417d225529c6d5ad24b7572815d0de9ac
	Date 2018-05-10T17:11:17.614Z
	Shell 1.7.12
	Renderer 58.0.3029.110
	Node 7.9.0
	Architecture x64

C# Extension version:

## Steps to reproduce

 1. Create console app
 2. Leave some error in the code and try to debug
 3. Click on the error in `Problems` tab

![image](https://user-images.githubusercontent.com/13226838/41198447-b9868704-6cbf-11e8-990f-ab395f446754.png)

## Expected  behavior

When I click on the error it should just take me to the error.

## Actual behavior

VSCode complains that the file cannot be found, even though it clearly knows about the file since it notified me about the error......

![capture](https://user-images.githubusercontent.com/13226838/41198440-77689ff6-6cbf-11e8-97f8-f2952db98f35.PNG)

**Comment by akshita31** [comment]
@sebasijan What is the C# extension version that you are using ?

**Comment by rchande** [comment]
@sebasijan I think this is a duplicate of https://github.com/OmniSharp/omnisharp-vscode/issues/1197

**Comment by delroh** [comment]
Have this same issue, but in my case I'm working with a aspnet core project and the file error is added twice:
1st entry is the one that when you click on it, shows the "file not found message". 
2nd entry, you click it and it takes you to the reported line and error. You fix it and the report dissappears, but 1st entry remains there and every time you click on it, you'll see this:

![image](https://user-images.githubusercontent.com/3700414/50038964-51917d00-0000-11e9-93cd-a28e29516cc1.png)

I can't make it dissapear unless I restart VSCode. The problem has already been fixed but this entry is there, cached with a non-existing reference.

### Environment Data
`dotnet --info` output:

```
.NET Core SDK (reflecting any global.json):
 Version:   2.1.502
 Commit:    c74ce8f29f

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.17134
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\2.1.502\

Host (useful for support):
  Version: 2.1.6
  Commit:  3f4f8eebd8
```

VSCode
```
Version: 1.30.0 (user setup)
Commit: c6e592b2b5770e40a98cb9c2715a8ef89aec3d74
Date: 2018-12-11T22:29:11.253Z
Electron: 2.0.12
Chrome: 61.0.3163.100
Node.js: 8.9.3
V8: 6.1.534.41
OS: Windows_NT x64 10.0.17134
```

**Comment by sebastianulm** [comment]
tldr; This needs to be in your tasks.json as build parameter for msbuild:
"/property:GenerateFullPaths=true"

Long story:
This is the default tasks.json that is generated (at least for me)
`{
    "version": "2.0.0",
    "tasks": [
        {
            "label": "build",
            "command": "dotnet",
            "type": "process",
            "args": [
                "build",
                "${workspaceFolder}<yourproject>.csproj",
            ],
            "problemMatcher": "$msCompile"
        }
    ]
}`

By default msbuild only generates relative paths. That's why you see the file:///project/file.cs not found error message. Omnisharp expects absolute paths (there is probably a reason for this silliness)
So add "/property:GenerateFullPaths=true" as a parameter to dotnet build should fix that problem.

**Comment by gregg-miskelly** [comment]
Alternatively, you can also change the problem matcher to '$tsc'. We changed the template in 1.18 to use '$tsc'.

**Comment by morden2k** [comment]
Any news about this issue? It is very uncomfortable go to file:line by manual file open with navigate to line instead double click to message.

How I see this issue exist on internet about 2 years already (((

Error is the same: `Unable to open 'XXX.cs': File not found ("file://XXX.cs").`
Where `XXX`- is file name without extension.

If I click to Create then it will be created in root folder (`/XXX.cs`) instead file path defined in project (e.g. `/home/qqq/some_path/XXX.cs`).

```
VS Code version: 1.31.1
OS: Ubuntu 14.04, Ubuntu 16.04
```

**Comment by djpnewton** [comment]
@morden2k have you tried the solution here: https://github.com/OmniSharp/omnisharp-vscode/issues/2374#issuecomment-452485460

**Comment by JoeRobich** [comment]
There have been lots of improvements since this issue was opened. Please open a new issue with logs if you are still running into this.
